File: faucet/rwdata/faucet/HandleCallServer.py
import json
import os
import pathlib
import requests


current_working_dir = str(pathlib.Path().absolute())
temp_data_dir = current_working_dir + '/tmp/'

class HandleCallServer:

    @staticmethod
    def get_server_url():
        config = {
            'serverUrl': 'http://192.168.43.176:8085'
        }
        if os.path.isfile(current_working_dir + '/config.json'):
            with open(current_working_dir + '/config.json', 'r') as f:
                config = json.load(f)
        return config['serverUrl']

    @staticmethod
    def get_rw_config():
        headers = {"Content-Type": "application/json",
                   "Accept": "application/json"}
        response = requests.get(url=HandleCallServer.get_server_url() + "/api/config/get-configrw", headers=headers)
        body = json.loads(response.text)
        print(body)
        return {"r": body["r"], "w": body["w"]}

    @staticmethod
    def get_version_from_server(ip):
        headers = {"Content-Type": "application/json",
                   "Accept": "application/json"}
        response = requests.get(url=HandleCallServer.get_server_url() + "/api/version/get-version?ip=" + ip,
                                headers=headers)
        body = json.loads(response.text)
        print(body)
        return body["version"]

    @staticmethod
    def update_version(ip, version):
        headers = {"Content-Type": "application/json",
                   "Accept": "application/json"}
        data = {"ip": ip, "version": version}
        response = requests.post(url=HandleCallServer.get_server_url() + "/api/version/update-version",
                                 data=json.dumps(data), headers=headers)
        print("update version to server: ", response.status_code)

    @staticmethod
    def send_log_read(data):
        headers = {"Content-Type": "application/json",
                   "Accept": "application/json"}
        try:
            response = requests.post(url=HandleCallServer.get_server_url() + "/api/log/log-read",
                                     data=json.dumps(data), headers=headers)
            print("send log read to server: ", response.status_code, response.text)
        except Exception as e:
            print(e)

    @staticmethod
    def send_log_write(data):
        headers = {"Content-Type": "application/json",
                   "Accept": "application/json"}
        try:
            response = requests.post(url=HandleCallServer.get_server_url() + "/api/log/log-write",
                                     data=json.dumps(data), headers=headers)
            print("send log write to server: ", response.status_code, response.text)
        except Exception as e:
            print(e)

File: onos/rwdata/src/main/java/org/onosproject/kcsg/locallistener/HandleCallServer.java
package org.onosproject.kcsg.locallistener;

import com.mashape.unirest.http.HttpResponse;
import com.mashape.unirest.http.Unirest;
import com.mashape.unirest.http.exceptions.UnirestException;

import org.json.JSONArray;
import org.json.JSONObject;
import org.onosproject.kcsg.locallistener.models.ConfigRWModel;

public final class HandleCallServer {
    private static String serverUrl = HandleVersion.getServerUrl();

    private HandleCallServer() {}

    public static ConfigRWModel getRWConfig() {
        try {
            HttpResponse<String> response = Unirest
                .get(serverUrl + "/api/config/get-configrw")
                .header("Content-Type", "application/json")
                .header("Accept", "application/json")
                .asString();

            JSONObject resBody = new JSONObject(response.getBody());

            ConfigRWModel config = new ConfigRWModel(resBody.getInt("r"), resBody.getInt("w"));
            return config;
        } catch (UnirestException e) {
            e.printStackTrace();
        }
        return null;
    }

    public static int getVersionFromServer(String ip) {
        try {
            HttpResponse<String> response = Unirest
                .get(serverUrl + "/api/version/get-version?ip=" + ip)
                .header("Content-Type", "application/json")
                .header("Accept", "application/json")
                .asString();

            JSONObject resBody = new JSONObject(response.getBody());
            return resBody.getInt("version");
        } catch (UnirestException e) {
            e.printStackTrace();
        }
        return 0;
    }

    public static JSONArray getVersionsFromServer() {
        try {
            HttpResponse<String> response = Unirest
                .get(serverUrl + "/api/version/get-versions")
                .header("Content-Type", "application/json")
                .header("Accept", "application/json")
                .asString();

            JSONArray resBody = new JSONArray(response.getBody());
            return resBody;
        } catch (UnirestException e) {
            //LOG.error(MSG, e.getMessage());
        }
        return null;
    }

    public static void updateVersion(JSONObject body) {
        try {
            HttpResponse<String> response = Unirest
                .post(serverUrl + "/api/version/update-version")
                .header("Content-Type", "application/json")
                .header("Accept", "application/json")
                .body(body)
                .asString();
        } catch (UnirestException e) {
            e.printStackTrace();
        }
    }

    public static void sendLogRead(JSONArray log) {
        try {
            HttpResponse<String> response = Unirest
                .post(serverUrl + "/api/log/log-read")
                .header("Content-Type", "application/json")
                .header("Accept", "application/json")
                .body(log)
                .asString();
        } catch (UnirestException e) {
            e.printStackTrace();
        }
    }

    public static void sendLogWrite(JSONArray log) {
        try {
            HttpResponse<String> response = Unirest
                .post(serverUrl + "/api/log/log-write")
                .header("Content-Type", "application/json")
                .header("Accept", "application/json")
                .body(log)
                .asString();
        } catch (UnirestException e) {
            e.printStackTrace();
        }
    }
}